s3torchconnector/src/s3torchconnector/_s3client/_s3client.py

```diff
@@ -54,12 +54,19 @@ def _client_builder(self) -> MountpointS3Client:
             region=self._region, user_agent_prefix=user_agent_prefix
         )
 
-    def get_object(self, bucket: str, key: str) -> S3Reader:
-        log.debug(f"GetObject s3://{bucket}/{key}")
+    def get_object(
+        self, bucket: str, key: str, *, object_info: Optional[ObjectInfo] = None
+    ) -> S3Reader:
+        log.debug(f"GetObject s3://{bucket}/{key}, {object_info is None=}")
+        if object_info is None:
+            get_object_info = partial(self.head_object, bucket, key)
```

Comment:
Just a thought, should we include debug logs in the if branches as well?

Reply:
What additional stuff would we log? I'm already logging if it takes the branch or not with the `object_info is None` logging

Reply:
Missed that, thanks!